mne/export/_edf.py

```diff
@@ -0,0 +1,210 @@
+
+    # set patient info
+    subj_info = raw.info.get('subject_info')
+    if subj_info is not None:
+        birthday = subj_info.get('birthday')
+        name = subj_info.get('first_name') + subj_info.get('last_name')
+        hand = subj_info.get('hand')
+        sex = subj_info.get('sex')
+
+        if birthday is not None:
+            if hdl.setPatientBirthDate(birthday[0], birthday[1],
+                                       birthday[2]) != 0:
+                raise RuntimeError(f"Setting Patient Birth Date to {birthday} "
+                                   f"returned an error")
+        for key, val in [('PatientName', name),
+                         ('PatientGender', sex),
+                         ('AdditionalPatientInfo', f'hand={hand}')]:
+            _try_to_set_value(hdl, key, val)
+
+    # set measurement date
+    meas_date = raw.info['meas_date']
+    if meas_date:
+        subsecond = meas_date.microsecond / 100.
+        if hdl.setStartDateTime(year=meas_date.year, month=meas_date.month,
+                                day=meas_date.day, hour=meas_date.hour,
+                                minute=meas_date.minute,
+                                second=meas_date.second,
+                                subsecond=subsecond) != 0:  # noqa
+            raise RuntimeError(f"Setting Start Date Time {meas_date} "
+                               f"returned an error")
+
+    device_info = raw.info.get('device_info')
+    if device_info is not None:
+        device_type = device_info.get('type')
+        _try_to_set_value(hdl, 'Equipment', device_type)
+
+    # Write each second (i.e. datarecord) separately.
+    for isec in range(np.ceil(n_secs).astype(int)):
+        end_samp = (isec + 1) * sfreq
+        if end_samp > n_times:
+            end_samp = n_times
+        start_samp = isec * sfreq
+
+        # then for each second write each channel
+        for ich in range(n_channels):
+            # create a buffer with sampling rate
+            buf = np.zeros(sfreq, np.float64, "C")
+
+            # get channel data for this second
+            ch_data = data[ich, start_samp:end_samp]
+
+            buf[:len(ch_data)] = ch_data
+            err = hdl.writeSamples(buf)
+            if err != 0:  # noqa
+                raise RuntimeError(f"writeSamples() returned error: {err}")
+
+        # there was an incomplete datarecord
+        if len(ch_data) != len(buf):
+            warn(f'A complete data record consists of {len(buf)} samples, '
+                 f'but this sample window ended up having {len(ch_data)} '
+                 f'samples. {len(buf) - len(ch_data)} zeros were appended '
+                 f'to the datarecord.')
+            warn(f'EDF format requires equal-length data blocks, '
+                 f'so {(len(buf) - len(ch_data)) / sfreq} seconds of zeros '
+                 f'were appended to all channels when writing the final '
+                 f'block.')
```

Comment:
two warnings, was that intentional?

Reply:
Ah no that was a git mess-up on my end. Fixed now and pushing.